drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -64,7 +63,6 @@ int lis2dh_reg_field_update(struct device *dev, u8_t reg_addr,
 	return lis2dh_reg_write_byte(dev, reg_addr,
 				     (old_val & ~mask) | ((val << pos) & mask));
 }
-#endif
```

Comment:
There will be a compilation warning though...
Is there a reason to do it? Or is it just for the sake to keep the code more logical?

Reply:
Which compilation warning excactly? It is not a static function, so there should be no such. This function will be thrown away on the link stage.

The main reason is to get rid of noisy #ifdef statements and maintaining them for all use cases. After the change we (developers) no longer need to think about every possible compilation option, but we will depend on linker to remove the function when it is really never used.

Reply:
ok, got it.